snl_shell/snl.c

```diff
@@ -38,15 +38,14 @@ int snl_forkpipe(char** args, char** args2);
 void snl_loop(void) {
   char *line;
   char *args[MAXARGS];
-  char *args2[MAXARGS];
   int status;
   int piped;
 
   do {
     printf("%s> ", get_cwd());
     line = snl_read_line();
-    piped = snl_detect_pipe(line, args, args2);
-    status = snl_execute(args, args2, piped);
+    snl_split_line(line, args); //TODO: idk about C practices but this kinda bothers me that it's not args = snl_split_line(line) - should they all be more uniform?
```

Comment:
We could change it to return a status of one like the other lines (think this is what Allen says is best practice). It's updating what args points to so it doesn't need to return it.